README.md

```diff
@@ -73,10 +79,23 @@ render(
 );
 ```
 
-**For a complete example of a fully-functional app using `react-prepare` in conjunction with `redux`, see the [react-prepare-todo](https://github.com/elierotenberg/react-prepare-todo) repository.**
-
 ### API
 
+#### `usePreparedEffect(sideEffect: async () => Promise<void>, deps, opts)`
+
+Works like `useEffect` except that when `prepare` is called, `sideEffect` is called (and awaited).
+
+Available `opts` is an optional configuration object:
+
+- `opts.awaitImmidiately` (default: `false`): on the server, `prepare` will await `sideEffect` before traversing further down the tree. When `false` the promise will be awaited before `prepare` returns.
+- `opts.runOnClient` (default: `true`): on the client, `sideEffect` is called inside a `useEffect` with the supplied dependency array.
```

Comment:
Idk, I don't really like `awaitOnSsr` because it awaits before ssr no matter what the option is set too. The difference is whether it awaits before running `prepare` on the child-components.

I thing `awaitInPrepare` would be better, but it's still kind on in `prepare` even if the option is `false`. Now that I think about about it something a little verbose like `awaitBeforePreparingChildren` might be ok, as the option is very rarely used anyways.

Reply:
I guess the way I reason about this might be a bit different. In my mind, the fact that it's awaited in `prepare` is a given, so `awaitInPrepare` in my mind, is that it is awaited while prepare runs (rather than after/the end), whidh is kind of the same as `awaitOnSsr`.

Something like `callSync` or something similar might work as well :shrug:. The other options you mentioned also work.